test/test_video_player.dart

```diff
@@ -0,0 +1,109 @@
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:video_player_platform_interface/video_player_platform_interface.dart';
+
+void setupMockVideoPlayerPlatform() {
+  VideoPlayerPlatform.instance = MockVideoPlayerPlatform();
+}
+
+class MockVideoPlayerPlatform extends Fake
+    with MockPlatformInterfaceMixin
+    implements VideoPlayerPlatform {
+  static const int _textureId = 0xffffffff;
+
+  static StreamController<VideoEvent> _streamController = StreamController<VideoEvent>();
+  static bool initialized = false;
+  static bool isPlaying = false;
+
+  static void reset() {
+    _streamController.close();
+    _streamController = StreamController<VideoEvent>();
+    initialized = false;
+    isPlaying = false;
+  }
+
+  @override
+  Future<void> init() async {}
+
+  @override
+  Future<void> dispose(int textureId) async {
+    assert(textureId == _textureId);
+    initialized = false;
+  }
+
+  @override
+  Future<int?> create(DataSource dataSource) async  {
+    initialized = true;
+    _streamController.add(VideoEvent(
+      eventType: VideoEventType.initialized,
+      duration: const Duration(seconds: 1),
+      size: const Size(0, 0),
+      rotationCorrection: 0,
+    ));
+    return _textureId;
+  }
+
+  @override
+  Stream<VideoEvent> videoEventsFor(int textureId) {
+    assert(textureId == _textureId);
+    return _streamController.stream;
+  }
+
+  @override
+  Future<void> setLooping(int textureId, bool looping) async {
+    assert(textureId == _textureId);
+  }
```

Comment:
Since we're not storing the `looping` value, let's check it has the value we expect, namely false.